test/test_perf.py

```diff
@@ -0,0 +1,172 @@
+
+  if args.compare_results and not args.old_results_path:
+    parser.error(
+        '--old_results_path and --compare_results must both be specified')
+
+  return args
+
+
+def run_driver_cmds(filename, install_path):
+  """Parse model file and launch Driver cmds"""
+  #tmp = filename.split('/')[-1]
+  resfile = f"{results_path}/{filename}"
+  model_path = f"{install_path}/share/miopen/perf_models/{filename}"
+
+  try:
+    outfile = open(os.path.expanduser(resfile), 'w+')
+    results = []
+    field_names = ['Driver', 'k_time']
+    writer = csv.DictWriter(outfile, fieldnames=field_names)
+    writer.writeheader()
+    with open(os.path.expanduser(model_path), "r") as infile:
+      for line in infile:
+        try:
+          idx = line.index('MIOpenDriver')
+          driver_cmd = line[idx:-1]
+          cmd = f"export LD_LIBRARY_PATH={install_path}/lib && export MIOPEN_LOG_LEVEL=6 && "\
+                f"{install_path}/bin/{driver_cmd} -V 0 -i 1 -w 1 -t 1"
+          print(f'Running cm: {cmd}')
+          p = subprocess.Popen(cmd,
+                               shell=True,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT)
+          p_out = p.stdout.readlines()
+          k_time = -1
+          for o_line in p_out:
+            o_line = o_line.decode("utf-8")
+            if 'GPU Kernel Time' in o_line:
+              split = re.search('Elapsed: (.*)ms', o_line)
+              k_time = split.group(1)
+            if 'error' in o_line:
+              raise ValueError(p_out)
+          results.append({'Driver': driver_cmd, 'k_time': k_time})
+          print(f'k_time: {k_time}')
+
+        except Exception as ex:
+          raise ValueError(f"Could not get kernel time: {ex}")
+
+    writer.writerows(results)
+    print(f"Perf results written to: {resfile}")
+    outfile.close()
+
+  except Exception as err:
+    outfile.close()
+    raise ValueError(f"Could not perform performance measurement: {err}")
+
+
+def compare_results(args):
+  """Compare current results with previous results"""
+  if not os.path.exists(results_path):
+    raise ValueError('Results path does not exist %s', results_path)
+  if not os.path.exists(args.old_results_path):
+    raise ValueError('Old results path does not exist %s',
+                     args.old_results_path)
+
+  if not compare_file(args, f"{results_path}/{args.filename}", \
+    f"{args.old_results_path}/{args.filename}"):
+    raise ValueError(f"FAILED: {args.filename}")
+  else:
+    print(f"PASSED: {args.filename}")
+
+
+def compare_file(args, new_results, old_results):
+  """Compare kernel_time in new vs old results file"""
+  with open(new_results, 'r') as new, open(old_results, 'r') as old:
+    for line_new, line_old in zip(csv.DictReader(new), csv.DictReader(old)):
+      if line_new['Driver'] != line_old['Driver']:
+        raise ValueError('Result files are out of sync')
+      if line_new['k_time'] > line_old['k_time']:
```

Comment:
[Q] The check does not have any tolerance. Can this lead to false failures?

Reply:
Probably so, what tolerance is acceptable?

Reply:
Our QA normally allows for 5% perf regressions, so I guess this is acceptable here as well.

Reply:
Resolved